nexusadspy/client.py

```diff
@@ -124,8 +124,11 @@ def _do_paged_get(self, url, method, params=None, data=None, headers=None,
     def _do_throttled_request(self, url, method, params=None, data=None, headers=None,
                               sec_sleep=2., max_failures=100,
                               get_field=None):
-
-        data = json.dumps(data)
+        try:
+            data = json.dumps(data)
+        except UnicodeDecodeError:
+            # json dump fails for byte stream. Pass data directly in that case.
```

Comment:
So it's a file-like object that's converted to a string by `json.dumps`, but `requests.request` accept file-like objects. 

Why can't you just pass the object directly to `requests.request` in all the cases?

Also I think you should pass different headers in this situations to specify the correct content type (`json.dumps(string) != string`).